### PR TITLE
x11rb-async: Use "tracing"

### PR DESCRIPTION
**Pull request by psychon**
This commit makes x11rb-async use the tracing crate. The tracing that is done is modelled after what is done in x11rb, but this time it is not optional, but a required dependency.

### CODE REVIEW

File: x11rb-async/examples/xclock_utc.rs [hunk 1]
```diff
@@ -316,14 +316,14 @@ async fn main2() -> Result<(), Box<dyn std::error::Error>> {
...
                             && data[0] == atoms.wm_delete_window
                         {
-                            println!("Window was asked to close");
+                            tracing::warn!("Window was asked to close");
                             return Ok::<_, Box<dyn std::error::Error>>(());
                         }
                     }
 
-                    Event::Error(_) => println!("Unexpected error"),
+                    Event::Error(e) => tracing::warn!("Unexpected error: {e:?}"),
```
Comment (notgull) [inline x11rb-async/examples/xclock_utc.rs:324]: Is this stable yet for our MSRV?

**Review by notgull — COMMENTED** [review]
Looks good to me!

File: x11rb-async/examples/xclock_utc.rs (hunk 1 above)
Comment (psychon) [reply x11rb-async/examples/xclock_utc.rs:324]: Nope. And "that other tracing PR" had a lot of problems with that, which is why I was careful with this one. Anyway, this is just an `example/` and we do not test those on our MSRV.

Actually... we do not even test x11rb-async against the MSRV. :see_no_evil:

**Comment by psychon** [comment]
I'll go ahead and merge this based on the "Looks good to me" above. Follow-up PR incoming.

(And the MSRV stuff in #821 annoys me currently...)

**Pull request merged**